Tasks/GooglePlayStatusUpdate/googleutil.ts

```diff
@@ -0,0 +1,169 @@
+// common code shared by all tasks
+import * as fs from 'fs';
+import * as tl from 'azure-pipelines-task-lib/task';
+import { google, androidpublisher_v3 } from 'googleapis';
+import { JWT } from 'google-auth-library';
+
+export const publisher: androidpublisher_v3.Androidpublisher = google.androidpublisher('v3');
+
+export interface ClientKey {
+    client_email?: string;
+    private_key?: string;
+}
+
+export interface AndroidRelease {
+    name?: string;
+    userFraction?: number;
+    releaseNotes?: ReleaseNotes[];
+    versionCodes?: [number];
+    status?: string;
+}
+
+export interface AndroidMedia {
+    body: fs.ReadStream;
+    mimeType: string;
+}
+
+export interface AndroidResource {
+    track?: string;
+    releases?: AndroidRelease[];
+}
+
+export interface Edit {
+    id: string;
+    expiryTimeSeconds: string;
+}
+
+export interface PackageParams {
+    packageName?: string;
+    editId?: string;
+    track?: string;
+    resource?: AndroidResource; // 'resource' goes into the 'body' of the http request
+    media?: AndroidMedia;
+    apkVersionCode?: number;
+    language?: string;
+    imageType?: string;
+    uploadType?: string;
+}
+
+export interface ReleaseNotes {
+    language?: string;
+    text?: string;
+}
+
+export interface Release {
+    name?: string;
+    versionCodes?: number[];
+    userFraction?: number;
+    releaseNotes: ReleaseNotes[];
+    status?: string;
+}
+
+export interface Track {
+    track: string;
+    releases: Release[];
+}
+
+export interface GlobalParams {
+    auth?: JWT;
+    params?: PackageParams;
+}
+
+export function getJWT(key: ClientKey): JWT {
+    const GOOGLE_PLAY_SCOPES: string[] = ['https://www.googleapis.com/auth/androidpublisher'];
+    return new google.auth.JWT(key.client_email, null, key.private_key, GOOGLE_PLAY_SCOPES, null);
+}
+
+/**
+ * Uses the provided JWT client to request a new edit from the Play store and attach the edit id to all requests made this session
+ * Assumes authorized
+ * @param {string} packageName - unique android package name (com.android.etc)
+ * @return {Promise} edit - A promise that will return result from inserting a new edit
+ *                          { id: string, expiryTimeSeconds: string }
+ */
+export async function getNewEdit(edits: any, globalParams: GlobalParams, packageName: string): Promise<Edit> {
```

Comment:
Could we use `androidpublisher_v3.Resource$Edits` type in all of these functions?
`getNewEdit`, `getTrack`, `updateTrack`

Reply:
Use androidpublisher_v3.Resource$Edits type in getNewEdit, getTrack, updateTrack, and remove unused type `Edit`, `Release`, `Track`.
Thanks